### ISSUE TITLE
Regression: application that works in v2.2.1 fails in v2.3.4 and later

### ISSUE THREAD

**Issue by RoxKilly** [Microsoft/TypeScript#16938]
**TypeScript Version:**  2.3.4 or 2.4.0

Upgraded an Angular project from `Typescript 2.0.3` to `2.3.4` and the previously working app now throws unexpected errors  at compile/design time.

**component.ts**
```ts
public tabs:Array<{
  title:string,
  description:string,
  data:Array<{name:string}>
}>=[];
```

**component.html**
```
<section *ngFor="let t of tabs">
  ...
  <div *ngFor="let i of t.data">{{i.name}}</div>
                                  ^^^^^^
</section>
```

**Compiler error (design time, in the IDE)**
> Angular: Identifier 'name' is not defined. `<anonymous>` does not contain such a member

If you open [this plunk](http://plnkr.co/edit/CZWAT3DKUo37EPx4KkP5?p=preview) you will notice that the application crashes (never loads). However if you go to line 44 of `config.js` and change the Typescript version to earlier (I've tried `2.2.1`), the application will load successfully

**Expected behavior:**

This code, which worked with earlier versions of TS should still work. The variables in question are strongly typed and well defined, so what's the problem?

**Comment by mhegazy** [comment]
I would report this on the agular issue tracker first. The TS compiler does not handle the template checking directly.

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
